Foundation/URLSession/URLSessionTask.swift

```diff
@@ -32,8 +32,30 @@ open class URLSessionTask : NSObject, NSCopying {
     
     /// How many times the task has been suspended, 0 indicating a running task.
     internal var suspendCount = 1
+    
+    internal var actualSession: URLSession? { return session as? URLSession }
```

Comment:
I was just wondering if we could create a property in `URLSessionProtocol` to provide the task delegate, some like below. This way we could keep the communication between `URLSessionTask` and `URLSession` through an interface instead of a concrete class. What do you think about this approach?

```swift
// URLSession.swift
internal protocol URLSessionProtocol: class {
    var taskDelegate: URLSessionTaskDelegate { get }
}
extension URLSession: URLSessionProtocol {
    var taskDelegate: URLSessionTaskDelegate? { return delegate }
}

// URLSessionTask.swift
func getBody(completion: @escaping (_Body) -> Void) {
    /* code */

    if let delegate = session.taskDelegate {
        /* code */
    }
}
```

Reply:
Honestly, the data is dispersed enough that it is an academic distinction.